docs/architecture/adr-038-state-listening.md

```diff
@@ -182,349 +137,587 @@ func (rs *Store) GetKVStore(key types.StoreKey) types.KVStore {
+       authtypes.StoreKey, banktypes.StoreKey, stakingtypes.StoreKey,
+       minttypes.StoreKey, distrtypes.StoreKey, slashingtypes.StoreKey,
+       govtypes.StoreKey, paramstypes.StoreKey, ibchost.StoreKey, upgradetypes.StoreKey,
+       evidencetypes.StoreKey, ibctransfertypes.StoreKey, capabilitytypes.StoreKey,
+    )
 
-e.g.
+    ...
+
+    // enable streaming
+    enableKey := fmt.Sprintf("%s.%s", baseapp.StreamingTomlKey, baseapp.StreamingEnableTomlKey)
+    if enable := cast.ToBool(appOpts.Get(enableKey)); enable {
+       pluginKey := fmt.Sprintf("%s.%s", baseapp.StreamingTomlKey, baseapp.StreamingPluginTomlKey)
+       pluginName := cast.ToString(appOpts.Get(pluginKey))
+       logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+       plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+       if err != nil {
+           tmos.Exit(err.Error())
+       }
+       if err := baseapp.RegisterStreamingPlugin(bApp, appOpts, keys, plugin); err != nil {
+           tmos.Exit(err.Error())
+       }
+    }
+
+    return app
+```
+
+#### Configuration
+
+The plugin system will be configured within an App's TOML configuration files.
 
 ```toml
-[plugins]
-    on = false # turn the plugin system, as a whole, on or off
-    enabled = ["list", "of", "plugin", "names", "to", "enable"]
-    dir = "the directory to load non-preloaded plugins from; defaults to "
-    [plugins.streaming] # a mapping of plugin-specific streaming service parameters, mapped to their plugin name
-        [plugins.streaming.file] # the specific parameters for the file streaming service plugin
-            keys = ["list", "of", "store", "keys", "we", "want", "to", "expose", "for", "this", "streaming", "service"]
-            write_dir = "path to the write directory"
-            prefix = "optional prefix to prepend to the generated file names"
-            halt_app_on_delivery_error = "false" # false == fire-and-forget; true == stop the application
-        [plugins.streaming.kafka]
-            keys = []
-            topic_prefix = "block" # Optional prefix for topic names where data will be stored.
-            flush_timeout_ms = 5000 # Flush and wait for outstanding messages and requests to complete delivery when calling `StreamingService.Close(). (milliseconds)
-            halt_app_on_delivery_error = true # Whether or not to halt the application when plugin fails to deliver message(s).
-        ...
+# gRPC streaming
+[streaming]
+
+# StreamingEnale defines if the gRPC streaming plugins should be enabled.
+enable = true
+
+# The plugin version to use for ABCI listening
+plugin = "grpc_abci_v1"
+
+# List of kv store keys to listen to for state changes.
+# Set to ["*"] to expose all keys.
+keys = ["*"]
+
+# Whether to stop the node on message deliver error.
+stop-node-on-err = true
 ```
 
+There will be four parameters for configuring the streaming plugin system: `streaming.enable`, `streaming.keys`, `streaming.plugin` and `streaming.stop-node-on-err`.
```

Comment:
nit: I would say `streaming.enable` is unnecessary. State streaming can be considered enabled by `streaming.plugin` being non-empty.

Reply:
removed.